devnotes/tpm.md

```diff
@@ -0,0 +1,228 @@
+
+| Aspect          | Scored   | Description                                              |
+| ------          | ------   | -----------                                              |
+| Interface       | No       | Interface of the chip: I2C, LPC, SPI                     |
+| Placement       | No       | Where TPM functionality is located physically            |
+| HW protected    | Yes      | Protection against tampering is by design                |
+| In stock        | Yes      | If modules compatible with the approach are easy to find |
+| Design HW       | Inverted | Designing hardware is necessary                          |
+| Manufacture HW  | Inverted | Producing hardware is necessary                          |
+| Needs OS driver | Inverted | Producing OS driver is necessary                         |
+| Fully open      | Yes      | Whether involved hardware and software are fully open    |
+
+## Approaches
+
+### I2C TPM module
+
+Hardware module with an uncommon interface (TPMs for PC mostly use LPC or SPI).
+Known chips of this kind are/were produced by Nuvoton, ST, Infineon,
+Atmel (AT97SC3205T).
+
+* **Pros**
+  * Talos II has I2C header exposed on the board.
+  * Best option if can be bought.
+
+* **Cons**
+  * No datasheets for Nuvoton chips.
+  * Limited or no availability of ready to use modules.
+  * Might require design and production of new TPM module with I2C interface.
+
+* **Risks**
+  * Hard to find and this might not improve in the future.
+
+* **Implementation effort**
+  * See [TPM over I2C](tpm_over_i2c.md).
+
+* **Other notes**
+  * ST33 has Linux driver for TPM 2.0
+  * Bought modules won't have OSS firmware.
+
+******
+### Software TPM on BMC
+
+TPM functionality exposed to the POWER9 CPU via LPC gadget device from BMC.
+As secure as BMC is.
+
+* **Pros**
+  * No additional hardware needed.
+  * Open implementation.
+
+* **Cons**
+  * Need to secure the BMC and manage that properly.
+
+* **Risks**
+  * BMC might not have been designed to fulfill security requirements.
+
+* **Implementation effort**
+  * Implement fTPM in BMC.
+  * LPC gadget BMC -> POWER.
+  * No existing driver. Maybe need to hack up existing LPC driver.
+
+* **Other notes**
+  * We could go full RO with coreboot if desired (by sacrificing some
+    performance for MVPD).
+
+******
+### FPGA translator (LPC TPM cycle - LPC TPM cycle)
+
+New piece of hardware that addresses LPC cycles issues.
+
+* **Pros**
+  * Conventional TPM modules.
+  * No need for a new driver.
+
+* **Cons**
+  * Need to design and manufacture this hardware.
+
+* **Risks**
+
+* **Implementation effort**
+  * Slightly easier than the fTPM on BMC - but not much.
+
+* **Other notes**
+
+******
+### FPGA translator (SPI - LPC TPM cycle)
+
+New piece of hardware that addresses LPC cycles issues.
+
+* **Pros**
+  * Conventional TPM modules.
+  * No need for a new driver.
+
+* **Cons**
+  * Need to design and manufacture this hardware.
+
+* **Risks**
```

Comment:
I don't think so: https://wiki.raptorcs.com/wiki/User:HLandau/Block_Diagram_Discussion#Minor_CPU_Interfaces

Reply:
@krystian-hebel @SergiiDmytruk Maybe this one was missing context. The SPI - LPC would use SPI TPM, not LPC TPM. So on the POWER9 side we would have the same situation as in case of the LPC-LPC translator. We would need to hack around the driver (?) to use I/O cycles to communicate with the SPI TPM over the SPI - LPC bridge.

This option was also interesting from the lpnTPM (http://lpntpm.lpnplant.io/) point of view. As in the first phase, we wil have SPI interface on the lpnTPM.

Reply:
@krystian-hebel what you would recommend here? Remove this option as not feasible by design?

Reply:
@krystian-hebel Please update based on my last comment.

Reply:
I believe this also can be done by @SergiiDmytruk

Reply:
I've already updated description to avoid the confusion.

Reply:
@krystian-hebel I'm not sure if you are around and there is any chance to merge it before end of week to keep the deadlines, but it would be great if you could look at it and if your concern was addressed then resolve the coversation - it would be one less before approval.

Reply:
Looks reasonable after clarification.